src/retro_data_structures/formats/strg.py

```diff
@@ -1,196 +1,145 @@
+    construct.Terminated,
 )
 
-StringTable = Struct(
-    "_start" / Tell,
-    If(this._.prime1, Seek(Int32ub.sizeof(), 1)),
-    "_size_start" / Tell,
-    "_offset_start" / Tell,
-    Seek(Int32ub.sizeof() * this._.string_count, 1),
-    "strings"
+STRG_V3 = Struct(
+    "header" / STRGHeader,
+    "name_table" / NameTable,
+    "language_ids" / Array(this.header.language_count, FourCC),
+    "language_table"
     / Array(
-        this._.string_count,
+        this.header.language_count,
         Struct(
-            "_size_start" / Tell,
-            "string" / CString("utf-16-be"),
-            "_size_end" / Tell,
-            "size" / Computed(this._size_end - this._size_start),
+            "strings_size" / Int32ub,
+            "string_offsets" / Array(this._.header.string_count, Int32ub),
         ),
     ),
-    "_size_end" / Tell,
-    "offsets" / Pointer(this._offset_start, StringTableOffsetAdapter(Int32ub)[this._.string_count]),
-    "_size" / Computed(this._size_end - this._size_start),
-    "size" / If(this._.prime1, Pointer(this._start, Rebuild(Int32ub, this._size))),
-)
-
-CorruptionString = Struct(
-    "_start" / Tell,
-    Seek(Int32ub.sizeof(), 1),
-    "_size_start" / Tell,
-    "string" / String,
-    "_size_end" / Tell,
-    "size" / Pointer(this._start, Rebuild(Int32ub, this._size_end - this._size_start)),
+    "strings" / construct.GreedyRange(Prefixed(Int32ub, CString("utf-8"))),
+    AlignTo(32),
+    construct.Terminated,
 )
 
-STRG = Struct(
-    "magic" / Const(0x87654321, Int32ub),
-    "version" / Enum(Int32ub, prime1=0, prime2=1, prime3=3),
-    "language_count" / Int32ub,
-    "string_count" / Int32ub,
-    "prime1" / Computed(this.version == "prime1"),
-    "prime2" / Computed(this.version == "prime2"),
-    "prime3" / Computed(this.version == "prime3"),
-    "lang_table_start" / Tell,
-    If(this.prime1 | this.prime2, Seek((FourCC.sizeof() + Int32ub.sizeof()) * this.language_count, 1)),
-    If(this.prime2, Seek(Int32ub.sizeof() * this.language_count, 1)),
-    "name_table" / If(this.prime2 | this.prime3, NameTable),
-    "corr_lang_ids_start" / Tell,
-    If(this.prime3, Seek(FourCC.sizeof() * this.language_count, 1)),
-    "corr_lang_table_start" / Tell,
-    If(this.prime3, Seek(Int32ub.sizeof() * (this.string_count + 1) * this.language_count, 1)),
-    "string_tables" / If(this.prime1 | this.prime2, StringTable[this.language_count]),
-    "string_table" / If(this.prime3, GreedyRange(CorruptionString)),
-    "string_table_length" / If(this.prime3, Computed(len_(this.string_table))),
-    "language_table" / If(this.prime1 | this.prime2, Pointer(this.lang_table_start, Language[this.language_count])),
-    "language_ids" / If(this.prime3, Pointer(this.corr_lang_ids_start, FourCC[this.language_count])),
-    "corruption_language_table"
-    / If(this.prime3, Pointer(this.corr_lang_table_start, CorruptionLanguage[this.language_count])),
-    "junk" / GreedyRange(Byte),
+STRG = construct.FocusedSeq(
```

Comment:
This isn't used anywhere right? it looks like `Strg.parse()`/`Strg.build()` makes it redundant.

Reply:
in `formats/__init__.py` it's exposed via the `format_for` and `has_format`. It's probably something we need to remove, but it's used in places, including directly in rdv. So uhhhhh

Reply:
ah, fair enough lol